### PR TITLE
Add invoices.credit endpoint

### PR DESCRIPTION
**Pull request by luisaterrabrito**

### CODE REVIEW

**Comment by luisaterrabrito** [comment]
I wonder if it makes sense to be invoices.credit and live in the credit notes, or if it should be creditNotes.creditInvoice. Or live in the invoices. What do you think?

File: apiary.apib [hunk 1]
```diff
@@ -2643,6 +2645,23 @@ Download a credit note in a specific format.
...
 
+
+### invoices.credit [POST /invoices.credit]
+
+Credit an invoice completely.
+
++ Request (application/json)
+
+    + Attributes (object)
+        + invoice_id: `d885e5d5-bacb-4607-bde9-abc4a04a901b` (string, required)
```
Comment (jenssegers) [inline apiary.apib]:
```suggestion
        + id: `d885e5d5-bacb-4607-bde9-abc4a04a901b` (string, required)
```

File: apiary.apib [hunk 2]
```diff
@@ -2643,6 +2645,23 @@ Download a credit note in a specific format.
...
+        + credit_note_date: `2016-02-04` (string, optional)
+
++ Response 200 (application/json)
+
+    + Attributes (object)
+        + data (object)
+            + credit_note_id: `d885e5d5-bacb-4607-bde9-abc4a04a901c` (string)
```
Comment (jenssegers) [inline apiary.apib]: Should return a type/id combination.

File: apiary.apib [hunk 3]
```diff
@@ -2643,6 +2645,23 @@ Download a credit note in a specific format.
...
+
+Credit an invoice completely.
+
++ Request (application/json)
+
+    + Attributes (object)
+        + invoice_id: `d885e5d5-bacb-4607-bde9-abc4a04a901b` (string, required)
+        + credit_note_date: `2016-02-04` (string, optional)
+
++ Response 200 (application/json)
```
Comment (jenssegers) [inline apiary.apib]:
```suggestion
+ Response 201 (application/json)
```

File: apiary.apib (hunk 1 above)
Comment (luisaterrabrito) [reply apiary.apib]: Living in the creditNotes can't it be confusing being just id?

**Review by jenssegers — DISMISSED** [review]

**Review by Yarn-e — DISMISSED** [review]

**Review by steftrenson — DISMISSED** [review]

**Review by Yarn-e — DISMISSED** [review]

**Review by jdrieghe — DISMISSED** [review]

File: src/changes-backwards-compatible.apib [hunk 4]
```diff
@@ -22,6 +22,8 @@ We list all backwards-compatible additions here. These are currently available i
 
 - We added `currency_exchange_rate` to `quotations.info`.
 
+- We've added a `invoices.credit` endpoint.
```
Comment (jdrieghe) [inline src/changes-backwards-compatible.apib]:
```suggestion
- We added an `invoices.credit` endpoint.
```

**Review by jdrieghe — APPROVED** [review]

**Review by Yarn-e — APPROVED** [review]

**Review by steftrenson — APPROVED** [review]

**Review by EOSullivanBerlin — APPROVED** [review]

**Review by jenssegers — APPROVED** [review]

**Pull request merged**